### PR TITLE
Add CMS_SA_ALLOW to Core classes

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

Added CMS_SA_ALLOW to mutables of classes which are never intended to be used across threads.
#### PR validation:

The code compiles.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

File: DataFormats/Provenance/interface/IndexIntoFile.h [hunk 1]
```diff
@@ -1077,7 +1080,8 @@ namespace edm {
       return transient_.eventFinder_->getEventNumberOfEntry(entry);
     }
 
-    mutable Transients transient_;
+    //This class is used only be one thread at a time within the source serialized code
```
Comment (makortel) [inline DataFormats/Provenance/interface/IndexIntoFile.h]:
```suggestion
    //This class is used only by one thread at a time within the source serialized code
```

**Comment by Dr15Jones** [comment]
please test

**Comment by Dr15Jones** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**